src/app/payment/services/payment.service.ts

```diff
@@ -82,6 +83,16 @@ const paymentService = {
     return paymentsClient.getPrices();
   },
 
+  async getCoupon(): Promise<CouponAvailable> {
+    const paymentsClient = await SdkFactory.getInstance().createPaymentsClient();
+    return paymentsClient.getCoupon();
+  },
+
+  async applyCoupon(): Promise<CouponAvailable> {
+    const paymentsClient = await SdkFactory.getInstance().createPaymentsClient();
+    return paymentsClient.applyCoupon();
```

Comment:
It could be less verbose if we do something like this: 

`return SdkFactory.getInstance().createPaymentsClient().applyCoupon();`

Reply:
To do it this way you would have to change createPaymentsClient() and SDK
It seems that this refactoring is out of the scope of the task.